geode-core/src/main/java/org/apache/geode/internal/net/NioSslEngine.java

```diff
@@ -248,12 +248,24 @@ public ByteBufferSharing wrap(ByteBuffer appData) throws IOException {
 
         SSLEngineResult wrapResult = engine.wrap(appData, myNetData);
 
-        if (wrapResult.getHandshakeStatus() == NEED_TASK) {
-          handleBlockingTasks();
+        switch (wrapResult.getStatus()) {
+          case BUFFER_UNDERFLOW:
+            throw new SSLException("Error encrypting data: " + wrapResult);
+          case BUFFER_OVERFLOW:
+            write(channel, myNetData);
+            break;
+          case OK:
+            break;
+          case CLOSED:
+            throw new SSLException("Error encrypting data: " + wrapResult);
         }
 
-        if (wrapResult.getStatus() != OK) {
-          throw new SSLException("Error encrypting data: " + wrapResult);
+        switch (wrapResult.getHandshakeStatus()) {
```

Comment:
Seems odd to evaluate this after some of the status above. Some of those should probably be returning rather than breaking. Could we end up with a `BUFFER_OVERFLOW` and `NEED_TASK`?

Reply:
this method changed significantly. resolving this conversation